instrumentctl/g9_driver.py

```diff
@@ -62,42 +54,115 @@ class G9Driver:
     }  
 
     def __init__(self, port=None, baudrate=9600, timeout=0.5, logger=None, debug_mode=False):
-        # not sure if we will need all of these but just adding them now in case
-        self.US, self.SITDF, self.SITSF, self.SOTDF, self.SOTSF, self.lastResponse, self.binSITDF, self.binSITSF = None, None, None, None, None, None, None, None
-
-        if port:  
-            self.ser = serial.Serial(  
-            port=port,  
-            baudrate=baudrate,  
-            parity=serial.PARITY_EVEN,  
-            stopbits=serial.STOPBITS_ONE,  
-            bytesize=serial.EIGHTBITS,  
-            timeout=timeout  
-            )  
-        else:  
-            self.ser = None
-
-        self.debug_mode = debug_mode
         self.logger = logger
+        self.debug_mode = debug_mode
+        self._setup_serial(port, baudrate, timeout)
+        self.last_data = None
         self.input_flags = []
 
-    def send_command(self):
+    def _setup_serial(self, port, baudrate, timeout):
+        if port:
+            try:
+                self.ser = serial.Serial(  
+                    port=port,  
+                    baudrate=baudrate,  
+                    parity=serial.PARITY_EVEN,  
+                    stopbits=serial.STOPBITS_ONE,  
+                    bytesize=serial.EIGHTBITS,  
+                    timeout=timeout  
+                    ) 
+                self.log(f"Serial connection established on {port}", LogLevel.INFO)
+            except serial.SerialException as e:
+                self.ser = None
+                self.log(f"Failed to open serial port {port}: {str(e)}", LogLevel.ERROR)
+        else:  
+            self.ser = None
+            self.log("No port specified", LogLevel.WARNING)
+
+    def get_interlock_status(self):
```

Comment:
The primary issue I saw with the earlier version of `send_command` was that it didn't actually return any status data when called in InterlocksSubsystem. The data received was stored as G9Driver class attributes rather than being returned to the caller. 

My understanding is that this violated OOP design conventions, and is a poor design because the caller has no way of knowing if the command actually succeeded. Storing the data as G9 class attributes and then accessing them directly from InterlocksSubsystem was also problematic because we'd have no way of knowing if and when the data becomes stale or when it was last updated.